dapp/src/components/page/proposal/ProposalStatusSection.tsx

```diff
@@ -1,26 +1,63 @@
-import React from "react";
-import type { ProposalViewStatus } from "types/proposal";
+import React, { useMemo } from "react";
+import { VoteType, type ProposalView } from "types/proposal";
 import { capitalizeFirstLetter } from "utils/utils";
 import { calculateDateDifference } from "../../../utils/formatTimeFunctions";
 
 interface Props {
-  status: ProposalViewStatus | null;
-  endDate: number | null;
+  proposal: ProposalView | null;
 }
 
-const ProposalStatusSection: React.FC<Props> = ({ status, endDate }) => {
+const ProposalStatusSection: React.FC<Props> = ({ proposal }) => {
+  const { status, voteResult, endDate } = useMemo(() => {
+    if (!proposal) return {};
+    const status = proposal.status;
+    const voteStatus = proposal.voteStatus;
+    let voteResult: VoteType | undefined = undefined;
+    const { approve, abstain } = voteStatus;
+    if (approve.score > abstain.score) {
+      voteResult = VoteType.APPROVE;
+    } else if (approve.score < abstain.score) {
+      voteResult = VoteType.REJECT;
+    } else {
+      voteResult = VoteType.CANCEL;
+    }
```

Comment:
You have duplicate codes with `dapp/src/components/page/proposal/VotingResult.tsx`